File: backend/src/main/java/com/rungo/api/global/infrastructure/mail/EmailService.java
package com.rungo.api.global.infrastructure.mail;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class EmailService {

    private final EmailSenderClient mailSenderClient;

    public void send(EmailMessage emailMessage) {
        mailSenderClient.send(emailMessage);
    }

    public void sendEmail(String to, String subject, String body) {
        send(new EmailMessage(to, subject, body));
    }
}

File: backend/src/main/java/com/rungo/api/global/infrastructure/mail/EmailService.kt
package com.rungo.api.global.infrastructure.mail

import org.springframework.stereotype.Service

@Service
class EmailService(
    private val mailSenderClient: EmailSenderClient
) {

    fun send(emailMessage: EmailMessage) {
        mailSenderClient.send(emailMessage)
    }

    fun sendEmail(to: String, subject: String, body: String) {
        send(EmailMessage(to, subject, body))
    }
}